samples/client/petstore/csharp-netcore/OpenAPIClient-generichost-net6.0-nrt/docs/models/DanishPig.md

```diff
@@ -4,7 +4,6 @@
 
 Name | Type | Description | Notes
 ------------ | ------------- | ------------- | -------------
-**ClassName** | **string** |  | 
```

Comment:
is it correct to remove `ClassName` from DanishPig?

```
    DanishPig:
      type: object
      properties:
        className:
          type: string
      required:
        - className
```

Reply:
Its the descriminator. We only need it for deserialization purposes. I don't see why it should be a property. When serializing, the value of the descriminator is inferred by the type. I think this is the right change.

Reply:
I restored the discriminator. I still think it is appropriate to treat it more like metadata where the discriminator value is populated based on the type. However, in the case of more complex (and probably questionable) specs, one model could appear in more than one oneOf/anyOf using different properties as discriminators. Handling that could get messy, so it is easier to just leave it like this.